src/video_core/buffer_cache/buffer_cache.cpp

```diff
@@ -251,6 +251,20 @@ u32 BufferCache::BindIndexBuffer(bool& is_indexed, u32 index_offset) {
         cmdbuf.bindIndexBuffer(stream_buffer.Handle(), offset, vk::IndexType::eUint16);
         return index_size / sizeof(u16);
     }
+    if (regs.primitive_type == AmdGpu::PrimitiveType::Polygon && !is_indexed) {
+        is_indexed = true;
+
+        // Emit indices.
+        const u32 index_size = 3 * regs.num_indices;
+        const auto [data, offset] = stream_buffer.Map(index_size);
+        Vulkan::LiverpoolToVK::EmitPolygonToTriangleListIndices(data, regs.num_indices);
+        stream_buffer.Commit();
+
+        // Bind index buffer.
+        const auto cmdbuf = scheduler.CommandBuffer();
+        cmdbuf.bindIndexBuffer(stream_buffer.Handle(), offset, vk::IndexType::eUint16);
+        return index_size / sizeof(u16);
+    }

```

Comment:
Think you should combine this into the previous QuadList case since all of the code is the same except the `EmitPolygonToTriangleListIndices` call, then just do a switch on the primitive type to call the right emit function.